core/cloudflow-runner/src/main/scala/cloudflow/runner/Runner.scala

```diff
@@ -68,17 +67,20 @@ object Runner extends RunnerConfigResolver with StreamletLoader {
         val streamletExecution = loadedStreamlet.streamlet.run(withPodRuntimeConfig)
         loadedStreamlet.streamlet.logStartRunnerMessage(formatBuildInfo)
 
-        Try {
-          // the runner waits for the execution to complete
-          // In normal circumstances it will run forever for streaming data source unless
-          // being stopped forcibly or any of the queries faces an exception
-          Await.result(streamletExecution.completed, Duration.Inf)
-        } match {
-          case Success(_) ⇒ System.exit(0)
-          case Failure(ex @ ExceptionAcc(exceptions)) ⇒
+        // the runner waits for the execution to complete
+        // In normal circumstances it will run forever for streaming data source unless
+        // being stopped forcibly or any of the queries faces an exception
+        Await.ready(streamletExecution.completed, Duration.Inf)
+        streamletExecution.completed.value match {
+          case None => // can't happen b/c we wait for it to be ready
+            log.error("Unexpected streamletExecution state: No result after termination. This is most probably a bug.")
+            shutdownWithFailure(loadedStreamlet,
+                                new IllegalStateException("Unexpected streamletExecution state: No result after termination"))
+          case Some(Success(_)) ⇒ System.exit(0)
```

Comment:
Maybe add a log that it is shutting down? Or is that already done by this point?